lib/SPIRVProducerPass.cpp

```diff
@@ -3818,15 +3818,80 @@ SPIRVID SPIRVProducerPass::GenerateInstructionFromCall(CallInst *Call) {
 
   switch (func_type) {
   case Builtins::kPopcount: {
-    //
-    // Generate OpBitCount
-    //
-    // Ops[0] = Result Type ID
-    // Ops[1] = Base ID
+    // The Base operand of any OpBitCount, OpBitReverse, OpBitFieldInsert,
+    // OpBitFieldSExtract, or OpBitFieldUExtract instruction must be a 32-bit
+    // integer scalar or a vector of 32-bit integers.
+    // https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/chap46.html#VUID-StandaloneSpirv-Base-04781
+    // Thus, non 32-bit types need to be managed explicitly.
+    const int BitcountSize = 32;
+    auto Ty = Call->getType();
+    auto BaseValue = Call->getOperand(0);
+    Type *Int32Ty = Ty->getInt32Ty(Context);
+    if (Ty->isVectorTy())
+      Int32Ty = VectorType::get(Int32Ty, dyn_cast<VectorType>(Ty));
+    auto TyBitWidth = Ty->getScalarSizeInBits();
+
     SPIRVOperandVec Ops;
-    Ops << Call->getType() << Call->getOperand(0);
+    if (TyBitWidth > BitcountSize) {
+      SPIRVID Value;
```

Comment:
```suggestion
      SPIRVID Value = getSPIRVValue(BaseValue);
```

Then you can drop the ifs to decide between using Value and BaseValue.